cmd/tendermint/commands/show_node_id.go

```diff
@@ -16,10 +19,16 @@ var ShowNodeIDCmd = &cobra.Command{
 }
 
 func showNodeID(cmd *cobra.Command, args []string) error {
-	nodeKey, err := p2p.LoadOrGenNodeKey(config.NodeKeyFile())
-	if err != nil {
-		return err
+	if cmn.FileExists(config.GenesisFile())&&cmn.FileExists(config.PrivValidatorFile()) {
```

Comment:
I think it should be more like 

```
nodeKey, err := p2p.LoadNodeKey(config.NodeKeyFile())
....
```

notice we use LoadNodeKey, not LoadOrGenNodeKey

Reply:
I think `LoadNodeKey `is a private function.

Reply:
ok, it's done

Reply:
you can remove this `if` now

Reply:
I was talking about `if cmn.FileExists(config.GenesisFile())&&cmn.FileExists(config.PrivValidatorFile()) {`!

Reply:
there is no need to check genesis and priv_validator files. LoadNodeKey will return an error if there is no node_key.json file.

Reply:
got it ^^